src/main/java/build/buildfarm/instance/memory/MemoryInstance.java

```diff
@@ -77,8 +78,8 @@ public MemoryInstance(String name, MemoryInstanceConfig config) {
     this(
         name,
         config,
-        /*contentAddressableStorage=*/ new HashMap<Digest, ByteString>(),
-        /*actionCache=*/ new HashMap<Digest, ActionResult>(),
+        /*contentAddressableStorage=*/ new ConcurrentHashMap<Digest, ByteString>(),
```

Comment:
tl;dr: actionCache/contentAddressableStorage population needs to be lifted out of constructor

I believe this betrays a larger interface problem here with usage vs. implementation.  The watchdog behavior (from operation-poll) represents concurrent manipulations of the structures under putOperation, but CAS and ActionCache are not represented in those manipulations.  Since the MemoryInstance does not perform other concurrent actions implicitly, the structures backing those two members needing to be concurrent is a function of a usage pattern (i.e. the multithreaded grpc handlers), and I'd rather not necessarily accommodate all usage patterns the implementation.